Google.PowerShell.IntegrationTests/Logging/Logging.GcLog.Tests.ps1

```diff
@@ -0,0 +1,102 @@
+﻿. $PSScriptRoot\..\GcloudCmdlets.ps1
+Install-GCloudCmdlets
+
+$project, $zone, $oldActiveConfig, $configName = Set-GCloudConfig
+
+Describe "Get-GcLogEntry" {
+    $r = Get-Random
+    $script:logName = "gcp-testing-log-$r"
+    $script:secondLogName = "gcp-testing-log2-$r"
+    $textPayload = "Test entry"
+    $secondTextPayload = "Second test entry"
+    $jsonPayload = "{\`"Key\`":\`"Value\`"}"
+    $timeBeforeCreatingLogs = [DateTime]::Now
+    gcloud beta logging write $logName $textPayload --severity=ALERT > $null
+    gcloud beta logging write $logName $jsonPayload --severity=INFO --payload-type=json
+    gcloud beta logging write $secondLogName $secondTextPayload --severity=ALERT
+    gcloud beta logging write $secondLogName $jsonPayload --severity=INFO --payload-type=json
+    # We add 2 minutes to account for delay in log creation
+    $timeAfterCreatingLogs = [DateTime]::Now.AddMinutes(2)
+
+    AfterAll {
+        gcloud beta logging logs delete $logName --quiet
+        gcloud beta logging logs delete $secondLogName --quiet
+    }
+
+    It "should work without any parameters" {
+        # There are a lot of logs so we just want to get the first 10
+        $logEntries = Get-GcLogEntry | Select -First 10
+        $logEntries.Count | Should Be 10
+    }
+
+    It "should work with -LogName parameter" {
+        $logEntries = Get-GcLogEntry -LogName $logName
+        $logEntries.Count | Should Be 2
+        $textLogEntry = $logEntries | Where-Object { -not [string]::IsNullOrWhiteSpace($_.TextPayload) }
+        $textLogEntry.TextPayload | Should BeExactly $textPayload
+        $jsonLogEntry = $logEntries | Where-Object { $null -ne $_.JsonPayload }
+        $jsonLogEntry.JsonPayload["Key"] | Should BeExactly "Value"
+
+        # Should not return anything for non-existent parameter
+        (Get-GcLogEntry -LogName "non-existent-log-name") | Should BeNullOrEmpty
+    }
+
+    It "should work with -Severity parameter" {
+        $logEntries = Get-GcLogEntry -Severity Alert
+        $logEntries.Count -ge 2 | Should Be $true
```

Comment:
We can't determine that because the entries can come from logs that we have no control over.